ec/src/lib.rs

```diff
@@ -274,10 +274,16 @@ pub trait AffineCurve:
     /// random group elements from a hash-function or RNG output.
     fn from_random_bytes(bytes: &[u8]) -> Option<Self>;
 
+    /// Multiplies `self` by the scalar represented by `bits`. `bits` must be a
+    /// big-endian bit-wise decomposition of the scalar.
+    fn mul_bits(&self, bits: impl Iterator<Item = bool>) -> Self::Projective;
```

Comment:
You can provide a default implementation of `mul_bits` that uses double-and-add.

Reply:
It doesn't work because `double_in_place` is not required in `AffineCurve`

Reply:
I also considered moving `mul_by_cofactor_to_projective` and `mul_by_cofactor_inv` into `AffineCurve` but it doesn't work because `AffineCurve` doesn't have the `P: ModelParameters` generic

Reply:
```suggestion
    fn mul_bits(&self, bits: impl Iterator<Item = bool>) -> Self::Projective {
        let mut res = Self::Projective::zero();
        // Skip leading zeros.
        for i in bits.skip_while(|b| !b) {
            res.double_in_place();
            if i {
                res.add_assign_mixed(&self)
            }
        }
        res
    }
```

Reply:
^ this should work

Reply:
But the `double_in_place`?

Reply:
`Self::Projective` should have a `double_in_place`, so this should work?

Reply:
Oooh I didn't read the error clearly, it says `multiple double_in_place found`